### PR TITLE
Fix virtualType for network-interfaces-scripted

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
Fix bug in #34258 and add a test ot prevent it

###### Things done
- [x] Tested via `nixos/tests/networking.nix`
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/tests/networking.nix [hunk 1]
```diff
@@ -433,6 +433,49 @@ let
...
+        $machine->start;
+        $machine->waitForUnit("network.target");
+
+        # Test interfaces set up
+        my $list = $machine->succeed("ip tuntap list | sort");
+        "$list" eq "$targetList" or die(
+          "The list of virtual interfaces does not match the expected one:\n",
+          "Result:\n", "$list\n",
+          "Expected:\n", "$targetList\n"
+        );
```
Comment (Mic92) [inline nixos/tests/networking.nix]: I would simplify the check to not depend on the output of ip:

```
$machine->succeed("ip link show tap0");
$machine->succeed("ip link show tun0"); 
```

Comment (rnhmjoj) [reply nixos/tests/networking.nix]: This ways it also checks the type and owner of the interface are set correctly.

**Comment by Mic92** [comment]
@GrahamcOfBorg test tests.networking.scripted.virtual

**Comment by Mic92** [comment]
@GrahamcOfBorg test networking.scripted.virtual

**Comment by rnhmjoj** [comment]
Woops, I forgot to squash the last commit.

**Pull request merged**